### ISSUE TITLE
DamageDealerWithEvent component breaks the inspector and throws errors

### ISSUE THREAD

**Issue by Leedotzip** [DanielEverland/ScriptableObject-Architecture#77]
Using Unity 2018.2.4f1 and the package from the releases tab, `Example #02 - Events.unity` uses the DamageDealerWithEvent script which is having issues being displayed inside the inspector and throwing console errors. 

`GUI Error: You are pushing more GUIClips than you are popping. Make sure they are balanced)`

`NullReferenceException: Object reference not set to an instance of an object
ScriptableObjectArchitecture.Editor.BaseReferenceDrawer.GetReferenceGenericType (UnityEditor.SerializedProperty property) (at Assets/SO Architecture/Editor/Drawers/BaseReferenceDrawer.cs:178)`

![example gif](https://i.gyazo.com/ae5e293d650db185d28c8bb47df524fd.gif)

**Comment by DanielEverland** [comment]
@jzapdot The issue is caused by an inheritance layer between `DamageDealerWithEvent` and `DamageDealer`, where a `BaseReference` type is in the base class, thereby causing a nullref in the following function.

```
public Type GetReferenceType(SerializedProperty property)
{
     return SerializedPropertyHelper.GetParent(property).GetType();
}
```

**Comment by DanielEverland** [comment]
Also, make sure to checkout master before looking into this, I've fixed a bug in BaseReferenceDrawer since v1.6.0

**Comment by DanielEverland** [comment]
Fixed
